shopfloor_reception_shipment_advice/services/reception.py

```
@@ -0,0 +1,99 @@
+# Copyright 2023 Camptocamp SA (http://www.camptocamp.com)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo.addons.component.core import Component
+
+
+class Reception(Component):
+    _inherit = "shopfloor.reception"
+
+    def _scan_document__by_dock(self, dock, barcode):
+        if not dock:
+            return
```

if you get there, a dock has been found, this looks unnecessary.
Plus, if an error is found (like you scanned a dock, but it is invalid in this context), it should return an error.

---

I followed the same logic that is in the `shopfloor_reception` module https://github.com/OCA/wms/blob/c1eca233cb2b19ee1fbb16eb258aa72fcd2d99a8/shopfloor_reception/services/reception.py#L345